docs/design.md

```diff
@@ -113,10 +113,18 @@ Drivers are essentially an extension of libnetwork and provides the actual imple
 * `driver.Join`
 * `driver.Leave` 
 
-These Driver facing APIs makes use of unique identifiers (`networkid`,`endpointid`,...) instead of names (as seen in user-facing APIs). 
+These Driver facing APIs makes use of unique identifiers (`networkid`,`endpointid`,...) instead of names (as seen in user-facing APIs).
 
 The APIs are still work in progress and there can be changes to these based on the driver requirements especially when it comes to Multi-host networking.
 
+### Driver semantics
+
+ * `Driver.CreateEndpoint`
+
+This method is passed an interface `EndpointInfo`, with methods `Interfaces` and `AddInterface`. If the slice returned by `Interfaces` is non-empty, the driver is expected to create the interfaces therein, and must return an error if it cannot. If the slice is empty, the driver should use `AddInterface` to provide zero or more fresh interfaces.
```

Comment:
The driver is expected to create the interfaces in the OS in any case, correct?

Maybe the last sentence should read "If the slice is empty, the driver should create one or more appropriate interfaces and use `AddInterface` to record their properties with the controller."

Note that I've changed this to "one or more" where you wrote "zero or more."  Is it ever appropriate for a driver to have an endpoint with zero interfaces?